app/src/main/java/eu/kanade/tachiyomi/ui/base/delegate/ThemingDelegate.kt

```diff
@@ -11,52 +11,25 @@ interface ThemingDelegate {
     fun applyAppTheme(activity: Activity)
 
     companion object {
-        fun getThemeResIds(appTheme: AppTheme, isAmoled: Boolean): List<Int> {
-            val resIds = mutableListOf<Int>()
-            when (appTheme) {
-                AppTheme.MONET -> {
-                    resIds += R.style.Theme_Tachiyomi_Monet
-                }
-                AppTheme.GREEN_APPLE -> {
-                    resIds += R.style.Theme_Tachiyomi_GreenApple
-                }
-                AppTheme.LAVENDER -> {
-                    resIds += R.style.Theme_Tachiyomi_Lavender
-                }
-                AppTheme.MIDNIGHT_DUSK -> {
-                    resIds += R.style.Theme_Tachiyomi_MidnightDusk
-                }
-                AppTheme.NORD -> {
-                    resIds += R.style.Theme_Tachiyomi_Nord
-                }
-                AppTheme.STRAWBERRY_DAIQUIRI -> {
-                    resIds += R.style.Theme_Tachiyomi_StrawberryDaiquiri
-                }
-                AppTheme.TAKO -> {
-                    resIds += R.style.Theme_Tachiyomi_Tako
-                }
-                AppTheme.TEALTURQUOISE -> {
-                    resIds += R.style.Theme_Tachiyomi_TealTurquoise
-                }
-                AppTheme.YINYANG -> {
-                    resIds += R.style.Theme_Tachiyomi_YinYang
-                }
-                AppTheme.YOTSUBA -> {
-                    resIds += R.style.Theme_Tachiyomi_Yotsuba
-                }
-                AppTheme.TIDAL_WAVE -> {
-                    resIds += R.style.Theme_Tachiyomi_TidalWave
-                }
-                else -> {
-                    resIds += R.style.Theme_Tachiyomi
-                }
-            }
+        private val themeResources: Map<AppTheme, Int> = mapOf(
+            AppTheme.MONET to R.style.Theme_Tachiyomi_Monet,
+            AppTheme.GREEN_APPLE to R.style.Theme_Tachiyomi_GreenApple,
+            AppTheme.LAVENDER to R.style.Theme_Tachiyomi_Lavender,
+            AppTheme.MIDNIGHT_DUSK to R.style.Theme_Tachiyomi_MidnightDusk,
+            AppTheme.NORD to R.style.Theme_Tachiyomi_Nord,
+            AppTheme.STRAWBERRY_DAIQUIRI to R.style.Theme_Tachiyomi_StrawberryDaiquiri,
+            AppTheme.TAKO to R.style.Theme_Tachiyomi_Tako,
+            AppTheme.TEALTURQUOISE to R.style.Theme_Tachiyomi_TealTurquoise,
+            AppTheme.YINYANG to R.style.Theme_Tachiyomi_YinYang,
+            AppTheme.YOTSUBA to R.style.Theme_Tachiyomi_Yotsuba,
+            AppTheme.TIDAL_WAVE to R.style.Theme_Tachiyomi_TidalWave,
+        )
 
-            if (isAmoled) {
-                resIds += R.style.ThemeOverlay_Tachiyomi_Amoled
+        fun getThemeResIds(appTheme: AppTheme, isAmoled: Boolean): List<Int> {
+            return buildList(2) {
+                add(themeResources.getOrDefault(appTheme, R.style.Theme_Tachiyomi))
```

Comment:
I don't know about this one, the `let` makes it longer than the current `add(...)` call. Also, I don't think the method reference is as readable. Especially because it tacks the actual logic being performed here on at the end, rather than at the beginning (in the same way the AMOLED overlay is added)

Reply:
Yeah it's fine